macros/calculate_age.yml

```
@@ -0,0 +1,20 @@
+version: 2
+
+macros:
+  - name: calculate_age
+    description: >
+      This macros calculates the accurate age in years based on a given date using the date_diff
+      function in BigQuery. The macros takes in two arguments date_1 and date_2. date_1 is the date of birth and by
+      default date_2 is current_date. Unless otherwise specified, the macros only calculates the current age when date_1
+      is provided. Upon providing date_2, the macros overrides the currrent_date and calculates the difference.
+      If only date_1 is provided, the current age is calculated. If date_1 and date_2 are provided, the age at death is
+      calculated.
```

```suggestion
      This macros calculates the age in years based on a given date using the date_diff function. The macros takes in two arguments:
      - patient_birth_date (required)
      - date_2 (optional) - the end date to calculate patient's age (e.g. date of decease, date of assigning a condition), with the current date as default
      
      If only patient_birth_date is provided, the patient's current age is calculated (as of today).
      If date_2 is provided, the patient's age at the specified date is calculated. 
```

I edited it to be briefer. One note is that you don't want to lock yourself in for the description of date_2 because it can be used for more things than just calculating a patient's age at decease, which is why we opted to generalize the function in the first place.